src/Command/CheckAllSchemaTemplatesNameFieldsCommand.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Jobcloud\SchemaConsole\Command;
+
+use Jobcloud\SchemaConsole\Helper\SchemaFileHelper;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class CheckAllSchemaTemplatesNameFieldsCommand extends Command
+{
+    private const TYPES_FOR_VALIDATION = [
+        'record',
+        'enum',
+        'fixed'
+    ];
+
+    private const REGEX_MATCH_FIRST_LETTER = '/^[A-Za-z_]/';
+
+    private const REGEX_MATCH_STRING = '/^[A-Za-z0-9_]+$/';
```

Comment:
You could incorporate the both rules in a single regex: REGEX_MATCH_NAME_NAMING_CONVENTION = `/^[a-z][a-z0-9_]*[a-z0-9]$/`
Ofc, we could also go a step further and ensure with a look behind that underscores are not repeating without at least a letter in between them.